### ISSUE TITLE
@types/node 13.5.0 breaks @feathersjs/transport-commons/lib/channels/channel/base.d.ts

### ISSUE THREAD

**Issue by petersaints** [feathersjs/feathers#1792]
### Steps to reproduce
I have a TypeScript project that uses the @feathersjs/socketio-client to connect to a Feathers server that has been working Ok. However, today I noticed a small problem while building it.

I was presented with the following error:
node_modules/@feathersjs/transport-commons/lib/channels/channel/base.d.ts:6:38 - error TS2507: 
```
> Type 'typeof EventEmitter' is not a constructor function type.
> 
> 6 export declare class Channel extends EventEmitter {
>                                        ~~~~~~~~~~~~
> 
> 
> Found 1 error.
```
After looking into the problem in detail I found out that it had to do with the EventEmitter declaration at https://github.com/DefinitelyTyped/DefinitelyTyped/commit/19826ba267c2203e61d28b87f56ca28eac23a994#diff-7d7d86e4d89d5a8d8e623c8873828549

It seems that EventEmitter is not a class anymore but a namespace which as an EventEmitter class on the latest version of @types/node (13.5.0)

I was able to fix the problem by reverting back to an old @types/node version (the latest v12.x.y version but v13.1.x also works). Which I actually think is what makes sense since I'm using the latest Node.js 12 LTS version. 

The v13.x.y of @types/node is only supposed to match the Node.js 13 API, right? If so, this is report is mostly a heads up so that you do the necessary changes to support future Node versions beyond v12 LTS.

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):
"@feathersjs/authentication": "^4.5.0",
"@feathersjs/authentication-client": "^4.5.0",
"@feathersjs/errors": "^4.5.0",
"@feathersjs/feathers": "^4.5.0",
"@feathersjs/socketio-client": "^4.5.0",
"@types/node": "^13.5.0",
"@types/socket.io-client": "^1.4.32"

**NodeJS version**:
12.14.1

**Operating System**:
Ubuntu 19.10 64-bit

**Browser Version**:
Firefox 73.0b9 Developer Edtion
**React Native Version**:
N/A
**Module Loader**:
TypeScript

**Comment by mlippert** [comment]
I just encountered this issue as well when I used the feathers cli to generate a new typescript app.

`@types/node` was not in the package.json dependencies that the cli created (I'm not sure I would have noticed the connection to the version of node I'm using, even if it was there).
In any case running:
```
npm install -D @types/node@^12
```
also fixed the issue for me. @petersaints thanks for the clear description of the issue and how to solve it.

**Comment by daffl** [comment]
This should be submitted as a bug in DefinitelyTyped. It is definitely still valid in the latest version of Node and it is a breaking change in a minor version.

**Comment by daffl** [comment]
Never mind, I stand corrected it was a wrong import that probably shouldn't have worked to begin with. Fixed in v4.5.1

**Comment by petersaints** [comment]
@daffl Thanks. I was ready to submit a report at DefinitelyTyped. I looked at the changed code on the PR and in fact the new way seems to be the correct even if you consider how things were on @types/node 13.1.8. I wonder how did it work before. But I only looked briefly at the diff I mentioned before (https://github.com/DefinitelyTyped/DefinitelyTyped/commit/19826ba267c2203e61d28b87f56ca28eac23a994#diff-7d7d86e4d89d5a8d8e623c8873828549) so it could be some nuance on the 13.1.8 file that allowed it to work the way it did.

@mlippert I'm glad I could help. I forgot to test with a brand new @feathers/cli project, but it's nice to have the confirmation that the problem wasn't being caused by some weird fluke on my own project. Also, my solution merely a work around but it's nice that know that it helped someone else.
